docs/content/4.sdk/2.getting-started/2.middleware-module.md

```diff
@@ -0,0 +1,471 @@
+
+```ts
+const product = await sdk.commerce.getProduct(
+  { id: "123" },
+  prepareConfig({
+    method: "GET",
+  })
+);
+```
+
+### Use a `GET` method by default
+
+You can use a `GET` method by default by passing it in the `defaultRequestConfig` option of the `middlewareModule` function.
+
+```ts
+export const sdk = initSDK({
+  commerce: buildModule(middlewareModule<UnifiedEndpoints>, {
+    apiUrl: "http://localhost:8181/commerce",
+    defaultRequestConfig: {
+      method: "GET",
+    },
+  }),
+});
+```
+
+### Add a header to each request
+
+You can add a header to each request by passing it in the `defaultRequestConfig` option of the `middlewareModule` function.
+
+```ts
+export const sdk = initSDK({
+  commerce: buildModule(middlewareModule<UnifiedEndpoints>, {
+    apiUrl: "http://localhost:8181/commerce",
+    defaultRequestConfig: {
+      headers: {
+        "X-Api-Key": "123",
+      },
+    },
+  }),
+});
+```
+
+### Add a header to single request
+
+You can add a header to a single request by passing it in the request configuration.
+
+```ts
+const product = await sdk.commerce.getProduct(
+  { id: "123" },
+  prepareConfig({
+    headers: {
+      "X-Custom-Header": "123",
+    },
+  })
+);
+```
+
+### Replace the default HTTP client with `axios`
+
+By default, the SDK uses the `fetch` API.
+You can replace the default HTTP client with `axios` by passing it in the `httpClient` option of the `middlewareModule` function.
+
+There are some requirements for the custom HTTP client.
+
+1. It must be an async function accepting the following parameters:
+   - `url` - the URL of the request,
+   - `params` - the parameters of the request,
+   - `config` - the request configuration,
+2. It must include the `Access-Control-Allow-Credentials` response header (more info [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Credentials)),
+3. It must throw an `SdkHttpError` if the request fails.
+
+```ts
+import axios from "axios";
+import { SdkHttpError } from "@vue-storefront/sdk";
+
+export const sdk = initSDK({
+  commerce: buildModule(middlewareModule<UnifiedEndpoints>, {
+    apiUrl: "http://localhost:8181/commerce",
+    httpClient: async (url, params, config) => {
+      try {
+        const { data } = await axios(url, {
+          ...config,
+          data: params,
+          withCredentials: true, // Includes the `Access-Control-Allow-Credentials` response header
+        });
+
+        return data;
+      } catch (error: any) {
+        throw new SdkHttpError({
+          statusCode: error.response?.status || 500,
+          message: error.response?.data?.message || error.message,
+          cause: error,
+        });
+      }
+    },
+  }),
+});
+```
+
+::: warning
+It's important to include the `Access-Control-Allow-Credentials` response header in the custom HTTP client. Otherwise, the cookies won't be sent with the request.
+:::
+
+### Add a custom error handler
```

Comment:
sidenote: we've got a similar guide for SAPCC refresh token
https://github.com/vuestorefront/unified-storefronts/blob/dfcdf703c1bfadf4309493976aee2d57e644947f/docs/content/3.features/5.authentication/9.refresh-token.md#implementing-the-token-refreshing-function

Reply:
Would be good to have a round through all of the docs and merge simillar pages into one